### PR TITLE
Analytics Aggregator V1

### PR DESCRIPTION
**Pull request by im-grahammartin**
This PR includes the initial MVP functionality for the analytics aggregator tool, incorporating our must-have requirements.

It includes: 
- Authenticating via the Google Analytics Data API to pull analytics data from multiple GA4 properties
- CLI interface for configuring report properties like dimensions, metrics and data ranges
- Data transformation and aggregation capabilities
- GA sites configured via code
- Results of reports saved as JSON exports
- Modular structure to keep GA API specific functionality co-located, and to enable improved maintenance and support for alternative interfaces and export options in future versions
- Dependencies managed via `pip`
- Comprehensive unit test coverage (98% coverage) using `pytest` and `pytest-cov`
- Basic Github Action to run unit tests on push or pull request

### CODE REVIEW

**Comment by im-grahammartin** [comment]
> It might be worth adding merge protections to the repository, to prevent unreviewed code merging into main.
> 
> Would be great to use this as an example for best practice for the data team possibly?

Agreed, this is a great opportunity to share some development DevOps practices across the wider team.

I had actually enabled the config but it was not enforced whilst the repository was still marked as private. I've now switched this to public and branch protections should now be enforced.

I've also responsed to your other comments on this PR, and amended the solution accordingly – thank you for the fast feedback!

File: src/helpers/aggregator.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                    if(key in aggregate[aggregate_key]):
+                        aggregate[aggregate_key][key] += result['metrics'][key]
+                    else:
+                        aggregate[aggregate_key][key] = result['metrics'][key]
+            else:
+                for key in result['metrics']:
+                    if(key in aggregate):
+                        aggregate[key] += result['metrics'][key]
+                    else:
+                        aggregate[key] = result['metrics'][key]
```
[the remark this answers is not in the document]
Comment (MKJSmith) [reply src/helpers/aggregator.py]: great to see you've raised the issue so it's not lost in the future

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import sys
+
+from helpers.arguments import get_arguments
+from helpers.ga_config import get_ga_properties
+from helpers.ga_api import generate_ga_report
+from helpers.aggregator import aggregate_reports
+from helpers.save_results import save_results
```
[the remark this answers is not in the document]
Comment (MKJSmith) [reply src/main.py]: The refactor is excellent; it's now much easier to see at a glance what components are at play to provide the logic for the application. 

Now, when someone wants to make modifications or additions, it's clearer where to apply them.

**Review by MKJSmith — APPROVED** [review]
Well structured with adequate test coverage for a new feature set.

**Pull request merged**